recovery/cmd/backup_size.go

```diff
@@ -0,0 +1,189 @@
+package cmd
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	disk "github.com/shirou/gopsutil/v3/disk"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	clusterPath         = "/var/lib/etcd/member/snap/db"
+	staticPodsPath      = "/etc/kubernetes/static-pod-resources/"
+	usrLocalPath        = "/usr/local"
+	kubeletPath         = "/var/lib/kubelet"
+	etcPath             = "/etc"
+	binPath             = "/etc/kubernetes/static-pod-resources/bin/"
+	backupSizeSafetyNet = 10.00
+	units               = []string{"B", "KiB", "MiB", "GiB", "TiB", "EiB", "ZiB"}
+)
+
+type resource struct {
+	dirPath string
+}
+
+type resourceList struct {
+	resources *[]resource
+}
+
+// compareBackupToDisk verifies disk space required against available
+// disk space and returns boolean value
+func compareBackupToDisk() (bool, error) {
+
+	estimated, err := estimateDiskSize()
+	if err != nil {
+		log.Errorf("Couldn't calculate estimated disk space")
+		return false, err
+	}
+
+	freeDisk, err := diskPartitionSize()
+	if err != nil {
+		log.Errorf("Couldn't calculate free disk space")
+		return false, err
+	}
+
+	estimated, ebytes := sizeConversion(estimated)
+	freeDisk, fbytes := sizeConversion(freeDisk)
+
+	log.Infof("Available disk space : %.2f %s; Estimated disk space required for backup: %.2f %s \n", freeDisk, fbytes, estimated, ebytes)
+
+	// find the index of estimated and freedisk space in units[]
+	var ebytesID, fbytesID int
+	for i := range units {
+		if units[i] == ebytes {
+			ebytesID = i
+		}
+		if units[i] == fbytes {
+			fbytesID = i
+		}
+	}
+
+	if ebytesID <= fbytesID {
+		// safetynet of 10GiB is added
```

Comment:
Comment could get stale by having the hardcoded value, which a reader could lookup. Or if this is configurable, could be inaccurate. I'd just remove the value and not that the size requirement is being padded

Reply:
I kept a a value for now which is hardcoded to 10GiB which can be changed based on our discussion but i was thinking if there is a case that after taking the backup there is no sufficient space left for the upgrade to continue, what will we do? The reason being introducing the safetyNet is to be sure we have some space left after the backup is taken.

Reply:
I'm just suggesting you drop "10GiB" from the comment here, as it could easily get outdated

Reply:
Dropped with commit effab4c9023ada38a70df52005e59d4bda2a8714